code/game/objects/buckling.dm

```diff
@@ -117,11 +117,71 @@
 /atom/movable/proc/post_unbuckle_mob(mob/living/M)
 	return
 
+/atom/movable/proc/user_buckle_minorfail(mob/living/M, mob/user)
+	M.forceMove(loc)
+	M.visible_message("<span class='notice'>[M] fails to pull [M.p_them()]self up onto [src].</span>",\
+		"<span class='warning'>You fail to climb onto [src]. Come on, you've got this!</span>",\
+		"<span class='italics'>You hear someone grunting.</span>")
+
+/atom/movable/proc/user_buckle_majorfail(mob/living/M, mob/user) //Knocks user out and does minor brute/brain damage
+	M.StopResting()
+	M.forceMove(loc) //All that effort wasn't for nothing!
+	M.visible_message("<span class='warning'>[M]'s hand slips and they slam their forehead on [src]!</span>",\
+		"<span class='danger'>SHIT! Your hand slips and you slam your forehead into [src]!</span>",\
+		"<span class='italics'>You hear a thwack and a heavy thump.</span>")
+	M.StartResting() //could if statement these lines down to playsound if you wanted this only to happen with certain seats
+	M.adjustBrainLoss(10)
+	M.Paralyse(5)
+	M.apply_damage(50, STAMINA)
+	M.apply_damage(15, BRUTE, "head")
+	M.AdjustSleeping(1)
+	playsound(src.loc, 'sound/items/trayhit1.ogg', 50, 0) //bonk
+
+//Allows people to climb onto chairs if they're next to or under them and incapacitated but not stunned
+//especially helpful for people with no legs. However it can fail on random chance based on how many hands they have
+/atom/movable/proc/user_buckle_incapacitated(mob/living/M, mob/user, check_loc = TRUE)
+	M.visible_message("<span class='warning'>[M] looks like they're attempting to pull [M.p_them()]self up onto [src].</span>",\
+		"<span class='warning'>You attempt to climb onto [src].</span>",\
+		"<span class='italics'>You hear someone grunting.</span>")
+	var/buckle_chance = rand(1, 100)
+	M.StartResting()
+	if (user.has_both_hands())
+		if(!in_range(user, src) || !isturf(user.loc) || M.anchored) //chair in range?
+			return FALSE
+		if (do_after(user, 2 SECONDS, target = src, use_default_checks=FALSE)) //Two hands
+			if (buckle_chance <= 5)
+				user_buckle_majorfail(M, user)
+				return FALSE
+			if (buckle_chance <= 25)
+				user_buckle_minorfail(M, user)
+				return FALSE
+	else if (do_after(user, 5 SECONDS, target = src, use_default_checks=FALSE)) //One hand
+		if(!in_range(user, src) || !isturf(user.loc) || M.anchored)
+			return FALSE
+		if (buckle_chance <= 20)
+			user_buckle_majorfail(M, user)
+			return FALSE
+		if (buckle_chance <= 75)
+			user_buckle_minorfail(M, user)
+			return FALSE
```

Comment:
Duplicate code!

I would define some variables beforehand, corresponding to the duration and fail chances depending on the user's presence of hands.

```dm
var/buckle_time = 2 SECONDS
var/minor_fail_chance = 5
var/major_fail_chance = 25
if(!user.has_both_hands())
	buckle_time = 5 SECONDS
	minor_fail_chance = 20
	major_fail_chance = 75

if(do_after(user, buckle_time, target = src, use_default_checks = FALSE))
	// ETC...
```

Also, the above numbers should be made into defines at the top of the file, as they are magic numbers.